### DISCUSSION TITLE
SPM installation?

### DISCUSSION THREAD

**Discussion by gemiren** [pointfreeco/swift-perception#5] [General]
I can't install it using SPM.

**Comment by valvoline** [comment]
You can. Just put the URL of the git and It'll work. I'm working with that, right now.

**Comment by gemiren** [comment]
That's the first thing I tried but it didn't work.

**Comment by mbrandonw** [comment]
Hi @gemiren, can you explain a bit more about what exactly didn’t work? It should work just like any SPM package.

**Comment by gemiren** [comment]
It stuck on "Verifying swift-perception.git...".

**Comment by stephencelis** [comment]
@gemiren Can you give detailed, step-by-step instructions on how we can reproduce? Including your version of Xcode, etc.?

**Comment by mbrandonw** [comment]
I'm going to convert this to a discussion because we don't think it's an issue with the library.

**Comment by gemiren** [comment]
Apparently it's a Xcode issue. After fighting Xcode with numerous resetting and rebooting I finally got the package added to SPM. Thanks for your help. Appreciated!

**Comment by davitykale** [comment]
@gemiren what did you have to do to get Xcode to recognize the library? I am stuck with a "No module 'Perception'" error

**Reply by mbrandonw** [reply]
You need to make sure to enable PerceptionMacros in Xcode before you can import the module. Usually there is an error in the issue navigator letting you know what macros need to be enabled, and when you click the error it asks you if you want to enable.
